aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -126,12 +129,21 @@ Tensor empty_strided_cpu(IntArrayRef size, IntArrayRef stride, const TensorOptio
   return t;
 }
 
-Tensor& empty_out(Tensor& result, IntArrayRef size) {
+Tensor& empty_out(
+    Tensor& result,
+    IntArrayRef size,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
+  TORCH_CHECK(
+      !(result.is_sparse() &&
+          optional_memory_format.has_value()),
+      " memory format options is incompatible with sparse tensors");
   check_size_nonnegative(size);
   if (result.is_sparse()) {
     result.sparse_resize_and_clear_(size, size.size(), 0);
   } else {
     result.resize_(size);
+    auto memory_format = optional_memory_format.value_or(MemoryFormat::Contiguous);
+    result.unsafeGetTensorImpl()->update_strides_to_format(memory_format);
```

Comment:
Umm... I know this is not your fault at all, but what the heck is `empty_out` supposed to do lol. It's not supposed to initialize the memory, so what's the point of it being `out=`? XD

Reply:
The comment I was originally going to write (but only makes sense for something like `ones_out`) is that if I pass in a tensor that is already strided (and the correct strides), I shouldn't be mucking around with the strides. So it seems to me that the logic here that defaults to `Contiguous` is not right, because I assume you're going to end up clobbering the old strides when you do this.

Reply:
I'm not sure it makes sense at all to accept memory format along with `out=`. I recommend erroring in this case.

Reply:
Agree, erroring out is the most clean solution. But fyi result.resize_(size) already mocking with strides.

Reply:
Not if the size matched to begin with. Which is what should be the case. (The most important use cases here are when result already has the correct size, and when result is empty)